core/src/main/scala/io/circe/sbt/CirceIoPlugin.scala

```diff
@@ -0,0 +1,85 @@
+package io.circe.sbt
+
+import scoverage._
+import org.scalastyle.sbt._
+import org.typelevel.sbt._
+import org.typelevel.sbt.gha._
+import sbt.Keys._
+import sbt._
+
+import GenerativeKeys._
+import TypelevelKernelPlugin._
+import autoImport._
+import TypelevelCiPlugin.autoImport._
+import TypelevelSonatypePlugin.autoImport._
+import ScalastylePlugin.autoImport._
+import ScoverageSbtPlugin.autoImport._
+
+object CirceIoPlugin extends AutoPlugin {
+  object autoImport
+
+  override def trigger = allRequirements
+
+  override def requires = TypelevelCiPlugin && TypelevelSonatypePlugin && ScalastylePlugin && ScoverageSbtPlugin
+
+  override def buildSettings =
+    publishSettings ++ organizationSettings ++ githubActionsSettings
+
+  override def projectSettings = Seq(
+    // TODO: look over projects and see what's in common
+  )
+
+  lazy val publishSettings: Seq[Setting[_]] =
+    Seq(
+      tlSonatypeUseLegacyHost := true
+    )
+
+  lazy val organizationSettings: Seq[Setting[_]] =
+    Seq(
+      organization := "io.circe",
+      organizationName := "io.circe"
+    )
+
+  lazy val githubActionsSettings: Seq[Setting[_]] = Seq(
+    githubWorkflowJavaVersions := List("8", "11", "17").map(JavaSpec.temurin(_)),
```

Comment:
Honestly I think all these Java versions is unnecessary. See this discussion; tl;dr http4s is entangled with TLS weirdness but Circe is not.
- https://github.com/http4s/sbt-http4s-org/issues/154

Reply:
Make sure to set `tlJdkRelease := Some(8)` then to make sure it still works with Java 8.